03-viz.Rmd

```diff
@@ -1,40 +1,528 @@
+print(islands_bar)
+```
+
+This is definitely clearer now, and allows us to answer our question ("are the top 7 largest landmasses continents?") in
+the affirmative. But the question could be made clearer from the plot by organizing the bars not by alphabetical order
+but by size, and to colour them based on whether or not they are a continent. In order to do this, we 
+use `mutate` to add a column to the data regarding whether or not the landmass is a continent:
+
+```{r 03-data-islands-bar-3}
+islands_top12 <- top_n(islands_df, 12, size)
+continents <- c('Africa', 'Antarctica', 'Asia', 'Australia', 'Europe', 'North America', 'South America')
+islands_ct <- mutate(islands_top12, is_continent = ifelse(landmass %in% continents, 'Continent', 'Other'))
+head(islands_ct)
+```
+
+In order to colour the bars, we add the `fill` argument to the aesthetic mapping. Then we use the `reorder` 
+function in the aesthetic mapping to organize the landmasses by their `size` variable.
+Finally, we use the `labs` and `theme` functions to add labels, change the font size, and position the legend:
+
+```{r 03-data-islands-bar-4}
+islands_bar <- ggplot(islands_ct, aes(x = reorder(landmass, size), y = size, fill = is_continent)) + 
+                   geom_bar(stat="identity") +
+                   labs(x = 'Landmass', y = 'Size (1000 square mi)', fill = 'Type') +
+                   coord_flip() +
+                   theme(text = element_text(size = 18), legend.position = c(0.75, 0.45))
+```
+
+```{r 03-data-islands-bar-4-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(islands_bar)
+```
+
+This is now a very effective visualization for answering our original questions. Landmasses are organized by their size,
+and continents are coloured differently than other landmasses, making it quite clear that continents are the largest 7 landmasses.
+
+
+### The Old Faithful eruption / waiting time dataset
+This dataset contains measurements of the waiting time between eruptions and the subsequent eruption duration (in minutes). **Question:** Is there a 
+relationship between the waiting time before an eruption to the duration of the eruption? 
+
+```{r 03-data-faithful}
+# old faithful eruption time / wait time data
+head(faithful)
+```
+
+Here again we are investigating the relationship between two quantitative variables (waiting time and eruption time). But if you look at the output of the `head` function, you'll notice that neither of the columns are ordered. So in this case, let's start again with a scatter plot:
+
+```{r 03-data-faithful-scatter}
+faithful_scatter <- ggplot(faithful, aes(x = waiting, y = eruptions)) + geom_point()  
+```
+```{r 03-data-faithful-scatter-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(faithful_scatter)
+```
+
+We can see that the data tend to fall into two groups: one with a short waiting and eruption times, and one with long waiting and eruption 
```

Comment:
Maybe spend another sentence here explaining this further in regards to the question you posed at the beginning of this sentence.